packages/p2p/src/components/my-ads/form-ads.jsx

```diff
@@ -37,26 +36,26 @@ class FormAds extends React.Component {
             rate: values.price_rate,
         };
         if (values.contact_info) {
-            request_payload.contact_info = values.contact_info;
+            create_advert.contact_info = values.contact_info;
         }
         if (values.payment_info) {
-            request_payload.payment_info = values.payment_info;
+            create_advert.payment_info = values.payment_info;
         }
         if (values.default_advert_description) {
-            request_payload.description = values.default_advert_description;
+            create_advert.description = values.default_advert_description;
         }
-        requestWS(request_payload).then(response => {
+        requestWS(create_advert).then(response => {
             // If we get an error we should let the user submit the form again else we just go back to the list of ads
-            if (response.error) {
-                this.setState({ error_message: response.error.message });
+            if (response.error && is_mounted) {
+                setErrorMessage(response.error.message);
                 setSubmitting(false);
             } else {
-                this.props.handleShowForm(false);
+                handleShowForm(false);
```

Comment:
`handleShowForm` updates state if I'm not mistaken, it should also happen inside an` is_mounted` check.